### DISCUSSION TITLE
USWDS should provide Angular components

### DISCUSSION THREAD

**Discussion by duluca** [uswds/uswds#4678] [Ideas]
## Description

Angular being the second most popular web framework, in the same vein as issue #2884, there needs to be Angular components of USDWS designs. Otherwise bringing in the current USWDS implementation results in gross inefficiencies, due to anti-patterns being introduced.

## What's the problem?

Government contracts often include inflexible language, preventing contractors from using anything but this specific USWDS implementation, even if a newer framework being used may implement or exceed the guidelines.

## Alternative Solution

At a minimum, README should be updated to recognize, frameworks like React and Angular may need alternative solutions (i.e. Angular Material) vs this particular implementation, so long as the spirit and general outlines of the guidelines are being adhered to.

**Comment by danizen** [comment]
It is also curious that uswds requires a peer component of react, if it is to support both.   I think the distribution of uswds needs to be split into at least:
* SCSS and minimal javascript
* React components

Possibly more, if this issue is to be worked.

**Comment by pinkfrogHK** [comment]
I implemented USWDS 2.0 component into Angular 7 by npm package, and when I run ng test, the error says uswds is not recognized. I ran gulp for running scss. what is folder structure of the folders or is there any instruction document for Angular set up?

**Comment by DaveSkender** [comment]
Recently, I've seen a few Angular/React projects that componentize the USWDS components:

- [GSA/ngx-uswds](https://github.com/GSA/ngx-uswds)
- [trussworks/react-uswds](https://github.com/trussworks/react-uswds)

Personally, I'm more excited to see this root upgrade of sources:

- #4986
